index.bs

```diff
@@ -4557,70 +4554,70 @@ Each {{ContextMenusEvent}} has an <dfn for=ContextMenusEvent>event listener list
 
   The <dfn method for=ContextMenusEvent>removeListener(|callback|)</dfn> method steps are:
 
-  1. Let |callbackList| be the [=ContextMenusEvent/event listener list=] of the {{ContextMenusEvent}}.
+  1. Let |callbackList| be the [=ContextMenusEvent/event listener list=] of [=this=].
 
-  1. If |callbackList|[|callback|] [=list/exists=], [=list/remove=] |callback| from |callbackList|.
+  1. [=list/For each=] |listItem| in |callbackList|:
+
+    1. If |listItem| equals to |callback|, [=list/remove=] |listItem| from |callbackList|.
 
 </div>
 
 <div algorithm>
 
   The <dfn method for=ContextMenusEvent>hasListener(|callback|)</dfn> method steps are:
 
-  1. Let |callbackList| be the [=ContextMenusEvent/event listener list=] of the {{ContextMenusEvent}}.
+  1. Let |callbackList| be the [=ContextMenusEvent/event listener list=] of [=this=].
+
+  1. [=list/For each=] |listItem| in |callbackList|:
 
-  1. If |callbackList|[|callback|] [=list/exists=], return `true`.
+    1. If |listItem| equals to |callback|, return `true`.
 
-  1. Otherwise, return `false`.
+  1. Return `false`.
 
 </div>
 
 <div algorithm>
 
   The <dfn method for=ContextMenusEvent>hasListeners()</dfn> method steps are:
 
-  1. Let |callbackList| be the [=ContextMenusEvent/event listener list=] of the {{ContextMenusEvent}}.
+  1. Let |callbackList| be the [=ContextMenusEvent/event listener list=] of [=this=].
 
   1. If |callbackList| [=list/is empty=], return `false`.
 
   1. Otherwise, return `true`.
 
 </div>
 
-<div algorithm>
-
-  The <dfn method for=ContextMenusEvent>dispatch(|data|)</dfn> method steps are:
-
-  1. Let |callbackList| be the [=ContextMenusEvent/event listener list=] of the {{ContextMenusEvent}}.
-
-  1. [=list/For each=] |callback| in |callbackList|:
-
-      1. Invoke |callback| with |data|.
-
-</div>
-
 {{ContextMenus}} has the attribute {{ContextMenus/onShow}}, which is a {{ContextMenusEvent}} that is associated with [= contextmenu =] [= events =].
 
 <div algorithm='onshow-event'>
 
-To [= dispatched =] an {{ContextMenus/onShow}} [= event =]:
+To [= dispatch =] an {{ContextMenus/onShow}} [= event =] |event|, given {{HTMLControlledFrameElement}} |controlledFrame|:
 
-1. Invoke {{ContextMenus/onShow}}'s {{ContextMenusEvent/dispatch(data)}} with no argument.
+1. Let |callbackList| be the [=ContextMenusEvent/event listener list=] of |event|.
+
+1. [=list/For each=] |callback| in |callbackList|:
+
+  1. 1. [=Queue a global task=] on the [=DOM manipulation task source=] of |controlledFrame|'s [=relevant global object=] to invoke |callback|.
```

Comment:
The double "1." is messing with the list rendering here, same on line 4663 and 4667